File: EssentialToolsCs/EssentialToolsCs/Models/ShoppingCart.cs
namespace EssentialToolsCs.Models
{
    using System.Collections.Generic;

    public class ShoppingCart
    {
        private readonly IValueCalculator calc;

        public ShoppingCart(IValueCalculator calcParam)
        {
            calc = calcParam;
        }

        public IEnumerable<Product> Products { get; set; }

        public decimal CalculateProductTotal()
        {
            return calc.ValueProducts(Products);
        }
    }
}

File: EssentialToolsVb0001/EssentialsToolsVb/Models/ShoppingCart.vb
Public Class ShoppingCart
    ReadOnly calc As IValueCalculator

    Public Sub New(calcParam As IValueCalculator)
        calc = calcParam
    End Sub

    Public Property Products() As IEnumerable(Of Product)

    Public Function CalculateProductTotal() As Decimal
        Return calc.ValueProducts(Products)
    End Function
End Class